### DISCUSSION TITLE
Wouldn't it be more appropriate to throw an exception?

### DISCUSSION THREAD

**Discussion by leeguooooo** [remix-run/remix#2492] [Proposals]
[Wouldn't it be more appropriate to throw an exception?](https://github.com/remix-run/remix/blob/621b392cc6151be51dc30f4e2bba4603085d5da5/packages/remix-server-runtime/cookies.ts#L187)

```javascript
function decodeData(value: string): any {
  try {
    return JSON.parse(atob(value));
  } catch (error) {
    return {};
  }
}
```

Wouldn't it be more appropriate to throw an exception or print it to the console?

It took me a long time to find the reason for this because of a bug.

Error in JSON.parse().

**Comment by kiliman** [comment]
Since Remix should be the one writing/encoding the cookie, the decode should never fail. It should only fail if somebody has tampered with the cookie. So instead of throwing an error, it simply ignores the invalid cookie and returns an empty object.

I believe there was an error with the `atob/btoa` function when using cookies with extended characters. There is a PR to fix this: #2257

**Comment by flameddd** [comment]
I still think we should add some logging when an error is caught.

The assumption that `Remix should be the one writing/encoding the cookie` is incorrect.

Cookies could be set by developers or other third-party libraries.

It’s difficult for beginners who are not familiar with how Remix handles cookies to debug without proper error logs.


I can open a PR for this, but I need some suggestion what kind of message we should leave ?
```js
function decodeData(value: string): any {
  try {
    return JSON.parse(decodeURIComponent(myEscape(atob(value))));
  } catch (error: unknown) {
    // some message like this ?
    console.error(`Can not parse following cookie value: ${value}`, error)
    return {};
  }
}
```

**Reply by kiliman** [reply]
You're using the cookie functions from Remix. These are not generic cookies.

If you want to manage your own cookies, then use any cookie package from npm, etc. You can even use the `cookie` package, which Remix uses. https://www.npmjs.com/package/cookie

I'm not saying I disagree with you. But it is what it is.

**Comment by flameddd** [comment]
I totally don’t understand the point of `... it is what it is.`. I did not suspect or complain about how Remix handles cookies.

What’s the main reason for refusing to add just one log message that could be helpful for developers? Why not ?

It's frustrated to debug/figure out issues when there is NO ANY message, and you to google/re-read document ... 

Is it considered too verbose?

I’m not trying to argue, just genuinely curious.

Thanks for everything.

**Reply by kiliman** [reply]
First of all, I'm not a core-contributor. I'm just pointing out how Remix currently works and the reason why. Feel free to submit a PR to get this changed. I was just pointing out that in the meantime, you can manage your own cookies and avoid this problem altogether.

Sorry if I sounded dismissive.

**Reply by flameddd** [reply]
I see. I will submit a PR for this. Thank you for your patience and explanation
